src/Kucipong/Handler/Store.hs

```diff
@@ -214,21 +217,28 @@ storeEditPost = do
       url
   redirect . renderRoute $ storeUrlPrefix
   where
-    -- TODO: put previous input texts
+    checkBusinessCategoryDetails :: BusinessCategory
+                                 -> [BusinessCategoryDetail]
+                                 -> ActionCtxT (HVect xs) m ()
+    checkBusinessCategoryDetails busiCat busiCatDets
+      | all (isValidBusinessCategoryDetailFor busiCat) busiCatDets = pure ()
+      | otherwise =
+        handleErr $ I18n.label def StoreErrorBusinessCategoryDetailIncorrect
+
     handleErr :: Text -> ActionCtxT (HVect xs) m a
     handleErr errMsg = do
+      p <- params
       $(logDebug) $ "got following error in storeEditPost handler: " <> errMsg
-      let
-        errors = [errMsg]
-        name = Nothing :: Maybe Text
-        businessCategory = Nothing :: Maybe BusinessCategory
-        businessCategoryDetails = [] :: [BusinessCategoryDetail]
-        salesPoint = Nothing :: Maybe Text
-        address = Nothing :: Maybe Text
-        phoneNumber = Nothing :: Maybe Text
-        businessHourLines = [] :: [Text]
-        regularHoliday = Nothing :: Maybe Text
-        url = Nothing :: Maybe Text
+      let errors = [errMsg]
+          name = lookup "name" p
```

Comment:
`Heterocephalus` can take default variables by giving variable name as `Ident` (`type Ident = Ident String`) and its value as `ExpQ`.
So, it can be another solution to declare new render function that takes dictionary as an argument, generates variables by given key names, and inject the variable into the template.